src/audio/kpb.c

```diff
@@ -129,13 +129,19 @@ static struct comp_dev *kpb_new(const struct comp_driver *drv,
 		.run = kpb_draining_task,
 		.get_deadline = kpb_task_deadline,
 	};
-	size_t bs = ipc_process->size;
 	struct comp_dev *dev;
 	struct comp_data *kpb;
 	int ret;
 
 	comp_cl_info(&comp_kpb, "kpb_new()");
 
+	/* make sure data size is not bigger than config space */
```

Comment:
@lgirdwood I suppose you added this to fix the DSP panic on the helios? But the problem is that the KPB expects the config data during new and doesnt handle it when passed via kcontrol IO

Reply:
This is during new(), it was never checked for size (and size was ignored).